src/metadata_utils.jl

```diff
@@ -72,19 +54,32 @@ function metadata_pkg(
     parentmodule(T).eval(ex)
 end
 
+# Extend `program` (an expression) to include trait definition for
+# specified `trait` and type `T`.
+function _extend!(program::Expr, trait::Symbol, value, T)
+    if value !== nothing
+        push!(program.args, quote
+              MLJModelInterface.$trait(::Type{<:$T}) = $value
+              end)
+    end
```

Comment:
nit: `push!` returns a value and so here the function may in some case return a vector of `eltype(program.args)` or sometimes nothing. Maybe just add a single `return` between the `end` to guarantee that the return type is always `Nothing` here?